### PR TITLE
feat(api): driver and simulator for FLEX Stacker

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
This PR adds a driver and a simulator for the FLEX Stacker that follow the same pattern as the other Opentrons modules.

The driver is in charge of sending commands to the device via serial communication, and parsing the responses into data that can be understood by the robot.

# What's Next?
I'll implement Stacker module control interface in the hardware controller layer that makes use of the driver.

### CODE REVIEW

File: api/src/opentrons/drivers/flex_stacker/driver.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    @classmethod
+    def parse_limit_switch_status(cls, response: str) -> LimitSwitchStatus:
+        """Parse limit switch statuses."""
+        field_names = LimitSwitchStatus.get_fields()
+        pattern = r"/s".join([rf"{name}:(?P<{name}>\d)" for name in field_names])
+        _RE = re.compile(f"^{GCODE.GET_LIMIT_SWITCH} {pattern}")
+        m = _RE.match(response)
+        if not m:
+            raise ValueError(f"Incorrect Response for limit siwtch status: {response}")
```
Comment (sfoster1) [inline api/src/opentrons/drivers/flex_stacker/driver.py]:
```suggestion
            raise ValueError(f"Incorrect Response for limit switch status: {response}")
```

**Review by sfoster1 — APPROVED** [review]
Excellent, thank you! One little typo, and we should add some tests, but the structure looks good

**Review by vegano1 — APPROVED** [review]
very nice!

**Review by sfoster1 — APPROVED** [review]
Nice, thank you!

**Pull request merged**